master/api_service.go

```diff
@@ -914,21 +914,33 @@ func (m *Server) getIPAddr(w http.ResponseWriter, r *http.Request) {
 
 func (m *Server) createMetaPartition(w http.ResponseWriter, r *http.Request) {
 	var (
+		vol     *Vol
 		volName string
-		start   uint64
+		count   int
 		err     error
 	)
 	metric := exporter.NewTPCnt(apiToMetricsName(proto.AdminCreateMetaPartition))
 	defer func() {
 		doStatAndMetric(proto.AdminCreateMetaPartition, metric, err, map[string]string{exporter.Vol: volName})
 	}()
 
-	if volName, start, err = validateRequestToCreateMetaPartition(r); err != nil {
+	if volName, count, err = validateRequestToCreateMetaPartition(r); err != nil {
 		sendErrReply(w, r, &proto.HTTPReply{Code: proto.ErrCodeParamError, Msg: err.Error()})
 		return
 	}
 
-	if err = m.cluster.updateInodeIDRange(volName, start); err != nil {
+	if vol, err = m.cluster.getVol(volName); err != nil {
+		sendErrReply(w, r, newErrHTTPReply(proto.ErrVolNotExists))
+		return
+	}
+
+	if err = vol.needCreateMetaPartition(); err != nil {
+		log.LogWarnf("action[createMetaPartition] no need to create meta partition , err[%v]", err)
+		return
```

Comment:
Forgot to send a reply?

Reply:
i just fixed this  problem and related unit tests